generic.go

```diff
@@ -41,14 +46,19 @@ func NewGeneric(config *Config, logger *zap.Logger, queue workqueue.RateLimiting
 		}
 		readyForWork := make(chan struct{})
 		queueGvk := wq.newQueueForGvk(descr.Gvk)
-		iface, err := constr.New(config, &Context{
-			ReadyForWork: func() {
-				close(readyForWork)
+		gvkLogger := logger.With(logz.Gvk(descr.Gvk))
+		iface, err := constr.New(
+			config,
+			gvkLogger,
```

Comment:
Hm, config contains the logger already. It would be confusing for the controller to get two loggers. Do we need two?

We should probably make a copy of the config here (just a shallow copy `x := *config`) and then put the `gvkLogger` into the copy and then pass it to the constructor. wdyt?

Reply:
I deleted the logger from the config because the logger is passed into the ControllerConstructor, with the Gvk field set. This should be the only logger used by the constructor. I could leave it in the config and do a deepcopy instead?